### PR TITLE
FCREPO-1713. Refactor EVERYONE user principal into the FAD.

### PR DESCRIPTION
**Pull request by peichman-umd**
Instead of a single static principal named EVERYONE in the ServletContainerAuthenticationProvider class, rely on each delegate to supply a Principal that it uses to represent the EVERYONE user through a getEveryonePrincipal() method.

Updated the integration tests with mocks for the EVERYONE principal for their mock FAD objects.

### CODE REVIEW

File: fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/ServletContainerAuthenticationProvider.java [hunk 1]
```diff
@@ -218,4 +195,13 @@ public void setFad(final FedoraAuthorizationDelegate fad) {
 
         return principals;
     }
+
+    /**
+     * Get the principal that the injected authorization delegate uses to represent the "EVERYONE" user.
+     *
+     * @return principal representing "EVERYONE"
+     */
+    public Principal getEveryonePrincipal() {
```
Comment (awoods) [inline fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/ServletContainerAuthenticationProvider.java]: Is this `public` method ever called? If not, please remove.

File: fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/ServletContainerAuthenticationProvider.java [hunk 2]
```diff
@@ -177,12 +154,12 @@ public ExecutionContext authenticate(final Credentials credentials,
         } else {
 
             sessionAttributes
-                    .put(FedoraAuthorizationDelegate.FEDORA_USER_PRINCIPAL,
-                            EVERYONE);
+            .put(FedoraAuthorizationDelegate.FEDORA_USER_PRINCIPAL,
```
Comment (awoods) [inline fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/ServletContainerAuthenticationProvider.java]: If you are going to reformat this line, please go ahead and move the `.put(...)` up to line:156.

File: fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/FedoraUserSecurityContext.java [hunk 3]
```diff
@@ -34,7 +32,7 @@
  * @author Gregory Jansen
  */
 public class FedoraUserSecurityContext implements SecurityContext,
-        AdvancedAuthorizationProvider {
+AdvancedAuthorizationProvider {
```
Comment (awoods) [inline fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/FedoraUserSecurityContext.java]: Generally, reformatting that has nothing to do with functional updates is frowned upon for the reason that it distracts the code review process, as well as makes code-history archaeology more difficult.

Comment (ajs6f) [inline fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/FedoraUserSecurityContext.java]: Just to be clear, this is entirely a concern of @awoods. No one else has ever so much as noticed it.

Comment (awoods) [inline fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/FedoraUserSecurityContext.java]: ...and @barmintor

Comment (peichman-umd) [inline fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/FedoraUserSecurityContext.java]: These formatting changes were put in there automatically by Eclipse. I would be happy to revert them.

**Comment by awoods** [comment]
Resolved with: https://github.com/fcrepo4/fcrepo4/commit/4629227735a6df3de25da2b8347e9cd190ef1c69

**Pull request closed without merging**